### PR TITLE
#76: Localize mobile app

### PR DESCRIPTION
**Pull request by mdokusV**

### CODE REVIEW

File: lib/main.dart [hunk 1]
```diff
@@ -21,12 +20,19 @@ import 'package:survey_frontend/presentation/screens/login_screen.dart';
...
 import 'package:survey_frontend/presentation/screens/welcome_screen.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+
 
 void main() async {
   await GetStorage.init();
   String startScreenPath =
       _getStartScreenPath(); 
   runApp(GetMaterialApp(
+    title: 'Localizations Sample App',
```
Comment (pawelFelcyn) [inline lib/main.dart:31]: What is this title?

File: lib/presentation/controllers/login_controller.dart [hunk 2]
```diff
@@ -47,31 +49,31 @@ class LoginController extends ControllerBase{
...
+      return AppLocalizations.of(Get.context!)!.invalidCredentials;
     }
 
     if (value == null || value == ''){
-      return 'Password must not be empty';
+      return AppLocalizations.of(Get.context!)!.passwordNotEmpty;
     }
 
     if (value.length > 255){
-      return 'Password must not be longer than 255 characters';
```
Comment (pawelFelcyn) [inline lib/presentation/controllers/login_controller.dart:58]: Can't this be somehow parametrized? For example, the translation is "Password must not be longer than $maxPasswordLen characters", and with Intl you can do something like:
```dart
return Intl.message("Password must not be longer than $maxPasswordLen characters",
args: [_maxPasswordLen]);
```
AFAIK

If the max length changes in the future, there will be only one place, that depends on it.

File: lib/presentation/controllers/login_controller.dart [hunk 3]
```diff
@@ -47,31 +49,31 @@ class LoginController extends ControllerBase{
...
+      return AppLocalizations.of(Get.context!)!.invalidCredentials;
     }
 
     if (value == null || value == ''){
-      return 'Username must not be empty';
+      return AppLocalizations.of(Get.context!)!.usernameNotEmpty;
     }
 
     if (value.length > 255){
-      return 'Username must not be longer than 255 characters';
```
Comment (pawelFelcyn) [inline lib/presentation/controllers/login_controller.dart:74]: Like the password error, this can be parametrized I believe.

**Review by pawelFelcyn — CHANGES_REQUESTED** [review]
In general it mostly looks good to me, but there is one more thing I don't like. Can't the localization files be somehow better structured? For example, it would be nice, if there would something like json-like structure, e. g.:
```
homeScreen: {
    "some text in home screen name": ""
},
"anotherScreen":{
    "some translation": ""
}
```

But I think:
- this is quite a lot to change
- the application is gonna be really small, so maybe this is not really needed

So if this is easy to do, then do it. But if we have to wait another week, then I'd maybe rather to merge this after the rest of the comments are resolved.

**Comment by mdokusV** [comment]
> In general it mostly looks good to me, but there is one more thing I don't like. Can't the localization files be somehow better structured? For example, it would be nice, if there would something like json-like structure, e. g.:
> ```
> homeScreen: {
>     "some text in home screen name": ""
> },
> "anotherScreen":{
>     "some translation": ""
> }
> ```
> 
> But I think:
> - this is quite a lot to change
> - the application is gonna be really small, so maybe this is not really needed
> 
> So if this is easy to do, then do it. But if we have to wait another week, then I'd maybe rather to merge this after the rest of the comments are resolved.

Looking at documentation unfortunately I can't find a quick way of doing it

File: lib/presentation/controllers/login_controller.dart (hunk 2 above)
Comment (mdokusV) [reply lib/presentation/controllers/login_controller.dart:58]: Yes, I'll change that

**Comment by pawelFelcyn** [comment]
> > In general it mostly looks good to me, but there is one more thing I don't like. Can't the localization files be somehow better structured? For example, it would be nice, if there would something like json-like structure, e. g.:
> > ```
> > homeScreen: {
> >     "some text in home screen name": ""
> > },
> > "anotherScreen":{
> >     "some translation": ""
> > }
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > But I think:
> > 
> > * this is quite a lot to change
> > * the application is gonna be really small, so maybe this is not really needed
> > 
> > So if this is easy to do, then do it. But if we have to wait another week, then I'd maybe rather to merge this after the rest of the comments are resolved.
> 
> Looking at documentation unfortunately I can't find a quick way of doing it

Ok, then let's leave this like it is. As I said, the app is gonna ve quite small, and most likely there is not a big risk it'll grow up. 

One more thing I forgot to mention. You are not loading a proper language based on settings, ot did I miss something?

File: lib/main.dart (hunk 1 above)
Comment (mdokusV) [reply lib/main.dart:31]: It apears in reasent opened apps. I'll change it to smth more suitable for our project

**Review by pawelFelcyn — APPROVED** [review]
now it looks god to me, I'll merge

**Pull request merged**